### PR TITLE
[vpp] Enable RGB565 to YUY2 color conversion

### PR DESCRIPTION
**Pull request by rkazants**
Successfully run sample_vpp for validation on SKL.

### CODE REVIEW

File: _studio/shared/src/libmfx_allocator_vaapi.cpp [hunk 1]
```diff
@@ -42,6 +42,11 @@
 #define VASurfaceAttribUsageHint 8
 #define VAEncMacroblockMapBufferType 29
 
+// TODO: remove this internal definition once it appears in VAAPI
+#define VA_FOURCC(ch0, ch1, ch2, ch3) \
```
Comment (lakulako) [inline _studio/shared/src/libmfx_allocator_vaapi.cpp]: MFX_MAKEFOURCC should be used. Isn't it the same?
What is the reason to introduce HERE local VA_FOURCC_R5G6B5? Isn't it defined elsewhere?

File: _studio/shared/src/libmfx_core.cpp [hunk 2]
```diff
@@ -1354,6 +1354,19 @@ mfxStatus CoreDoSWFastCopy(mfxFrameSurface1 *pDst, mfxFrameSurface1 *pSrc, int c
 
         break;
 
+#if (MFX_VERSION >= MFX_VERSION_NEXT)
+    case MFX_FOURCC_RGB565:
+        {
+            mfxU8* ptrSrc = MFX_MIN(MFX_MIN(pSrc->Data.R, pSrc->Data.G), pSrc->Data.B);
```
Comment (lakulako) [inline _studio/shared/src/libmfx_core.cpp]: That is why  R G B should be equal: to get rid of this unclear operations.

File: samples/sample_common/src/sysmem_allocator.cpp [hunk 3]
```diff
@@ -130,6 +130,13 @@ mfxStatus SysMemFrameAllocator::LockFrame(mfxMemId mid, mfxFrameData *ptr)
         ptr->V = ptr->Y + 3;
         ptr->Pitch = 2 * Width2;
         break;
+#if (MFX_VERSION >= MFX_VERSION_NEXT)
+    case MFX_FOURCC_RGB565:
+        ptr->G = ptr->B;
+        ptr->R = ptr->B + 1;
```
Comment (lakulako) [inline samples/sample_common/src/sysmem_allocator.cpp]: The same comments as above for the similar code below.

File: _studio/shared/src/libmfx_allocator_vaapi.cpp (hunk 1 above)
Comment (rkazants) [reply _studio/shared/src/libmfx_allocator_vaapi.cpp]: VA_FOURCC_R5G6B5 is not defined in VA API.

**Comment by rkazants** [comment]
Fixed Leonid's comments

File: _studio/shared/src/libmfx_core.cpp (hunk 2 above)
Comment (lakulako) [reply _studio/shared/src/libmfx_core.cpp]: const mfxU8* ptrSrc = pSrc->Data.B;
mfxU8* ptrDst = pDst->Data.B;

**Comment by rkazants** [comment]
Fixed two Leonid's comments yet.

**Pull request merged**